### DISCUSSION TITLE
gopro-cut duration syntax

### DISCUSSION THREAD

**Discussion by DougEdwards47** [time4tea/gopro-dashboard-overlay#197] [General]
What is the exact syntax for specifying the duration (or start or end). I have tried:
 --duration 00:10:10.000000
  --duration (00:10:10.000000)
   --duration \(00:10:10.000000\)
    --duration "00:10:10.000000"
    A typical error message is:
    Traceback (most recent call last):
  File "/usr/local/bin/venv/bin/gopro-cut.py", line 33, in <module>
    from_seconds = as_seconds(args.start)
                   ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/bin/venv/bin/gopro-cut.py", line 13, in as_seconds
    parsed_time = parse_time(timey)
                  ^^^^^^^^^^^^^^^^^
  File "/usr/local/bin/venv/lib/python3.12/site-packages/gopro_overlay/parsing.py", line 7, in parse_time
    return datetime.strptime(s, f)
           ^^^^^^^^^^^^^^^^^^^^^^^
TypeError: strptime() argument 1 must be str, not None

**Comment by time4tea** [comment]
You should be able to pass in any of the following:
`10`   - 10 seconds
`10.2` - 10.2 seconds
`1:10`   - 1 min 10 seconds
`1:10.2` - 1 min 10.2 seconds
`1:1:3` - 1 hour 1 minutes 3 seconds
`1:1:3.2` - 1 hour 1 minutes 3.2 seconds

it seems to be complaining about the value for `--start` though, not duration?
